### PR TITLE
Exposing last edited property of Entity to Javascript

### PR DESCRIPTION
**Pull request by fayeli**
Builds fine with no error but last edited is still not available as a property through script. Love to get some guidance on this! thanks!

### CODE REVIEW

**Comment by thoys** [comment]
Wasn't it exposed already?  `Entities.getEntityProperties('{35853841-0b38-4735-9db1-41d7f48fe807}').lastEditedBy`   or  is lastEdited a new property with a datetime?

**Comment by thoys** [comment]
Ah I see, its a new property, nevermind :)

**Comment by thoys** [comment]
@fayeli this is what Seth did to add the lastEditedBy property, might help: https://github.com/highfidelity/hifi/commit/850361f22233501f7bf0bbcd2c958be4485d4c51

File: libraries/entities/src/EntityPropertyFlags.h [hunk 1]
```diff
@@ -182,6 +182,7 @@ enum EntityPropertyList {
     PROP_LOCAL_ANGULAR_VELOCITY, // only used to convert values to and from scripts
 
     PROP_LAST_EDITED_BY,
+	PROP_LAST_EDITED,
```
Comment (ZappoMan) [inline libraries/entities/src/EntityPropertyFlags.h]: formatting is messed up

File: libraries/entities/src/EntityItemProperties.cpp [hunk 2]
```diff
@@ -369,6 +369,7 @@ QScriptValue EntityItemProperties::copyToScriptValue(QScriptEngine* engine, bool
         COPY_PROPERTY_TO_QSCRIPTVALUE_GETTER_NO_SKIP(ageAsText, formatSecondsElapsed(getAge())); // gettable, but not settable
     }
 
+	COPY_PROPERTY_TO_QSCRIPTVALUE(PROP_LAST_EDITED, lastEdited);
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItemProperties.cpp]: formatting is messed up

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0fcb8803b6a28b2dbe4ef13810b50b35decb5187 available at: https://gist.github.com/b99e0b7b468d352b572d5327886e6756<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-0fcb880.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-0fcb880.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b6798adb5c823ca3963c04eadabeb88949aad66b available at: https://gist.github.com/b99e0b7b468d352b572d5327886e6756<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-b6798ad.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-b6798ad.dmg

**Comment by fayeli** [comment]
Test Plan:

1) Run Interface from this PR
2) Create an Entity (can be sphere, box or whatever). Copy its ID
3) Open Console, run 
                   var props = Entities.getEntityProperties(ID);
                   JSON.stringify(props);
4) Console should return all properties of your entity. Confirm that there is a field "lastEdited" which should just be a big number in milliseconds in UNIX time
5) Now, edit you entity (e.g. change its color) and rerun step 3) in console
6) Confirm the new "lastEdited" is now updated to a new time.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1fde68c86e5deb57d4a6f1476a03b271ea075511 available at: https://gist.github.com/b99e0b7b468d352b572d5327886e6756<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-1fde68c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-1fde68c.dmg

**Comment by Polyrhythm** [comment]
Instead of having manual QA steps for testing this new endpoint, it's more maintainable to include a unit test.

You can add test coverage for this property by adding a unit test to [the runner](https://github.com/highfidelity/hifi/blob/master/scripts/developer/tests/unit_tests/testRunner.js) or directly to the [entity unit tests](https://github.com/highfidelity/hifi/blob/master/scripts/developer/tests/unit_tests/entityUnitTests.js). It would be nice to default to adding coverage to all new api endpoints.

Run by Developer -> Tests -> Run Client Script Tests

**Comment by OmegaHeron** [comment]
Tested Windows 10.

This functions as described.  I'm unclear on what its actual use is since last edit is only in scope of last edit within this run of Interface...  If you leave domain/return and call function, last edit = now -- restart Interface and call, last edit = now.  So it's really last edited during this visit to this domain during this run of Interface -- not last edited globally.

General smoke test pass.

**Comment by sethalves** [comment]
If someone else edits an entity, does lastEdited change for the observer?

**Comment by ZappoMan** [comment]
@sethalves - it should work ... basically if someone else edits the entity (and you're looking at it) then you'll get an update from the server, that update will include the new last edited time.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9289dd36d4a7e2d40b4a94ba2512d243d423d554 available at: https://gist.github.com/b99e0b7b468d352b572d5327886e6756<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-9289dd3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-9289dd3.dmg

**Comment by fayeli** [comment]
Thanks for manually testing this PR @Polyrhythm @OmegaHeron !

I've just added a unit test as suggested. Let me know if that's working fine for you! Thanks!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1644c2eeb43ab7d2a01fe275655c5ec8e06dc0e3 available at: https://gist.github.com/b99e0b7b468d352b572d5327886e6756<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-1644c2e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-1644c2e.dmg

**Comment by OmegaHeron** [comment]
Re-tested Windows 10

Manual tests - pass
Automated unit tests - pass

unit-test-pr9164

**Comment by Polyrhythm** [comment]
👍

**Comment by hifi-gustavo** [comment]
Build console output for commit: b3ac62bb56b8fbec397dd50625e7304fe1f8e921 available at: https://gist.github.com/b99e0b7b468d352b572d5327886e6756<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-b3ac62b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9164-b3ac62b.dmg

**Pull request merged**